File: web/src/lib/type.ts
export interface Board {
	row: number;
	col: number;
	isLight: boolean;
	position: string;
	piece: string | null; // e.g. "wP" for white pawn, "bK" for black king
}

export enum pieceType {
	Pawn = "P",
	Knight = "K",
	Bishop = "B",
	Rook = "R",
	Queen = "Q",
	King = "K",
}

File: go/engine/type.go
package engine

type Color int

const (
	White Color = iota
	Black
)

type PieceType string

const (
	Pawn   PieceType = "P"
	Knight           = "N"
	Bishop           = "B"
	Rook             = "R"
	Queen            = "Q"
	King             = "K"
)

type Piece struct {
	Type  PieceType
	Color Color
}

type Move struct {
	From    string
	To      string
	IsToEat bool
}

type Board struct {
	Row      int     `json:"row"`
	Col      int     `json:"col"`
	IsLight  bool    `json:"isLight"`
	Position string  `json:"position"`
	Piece    *string `json:"piece"`
}
